src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java

```diff
@@ -667,22 +672,22 @@ public CompletableFuture<FlowRerouteResponseV2> rerouteFlowV2(String flowId) {
     }
 
     @Override
-    public CompletableFuture<List<FlowHistoryEntry>> listFlowEvents(String flowId,
-                                                                    long timestampFrom,
-                                                                    long timestampTo, int maxCount) {
+    public CompletableFuture<List<FlowHistoryEntry>> getFlowHistory(String flowId,
+                                                                    FlowHistoryRangeConstraints constraints) {
         log.info("API request: List flow events: flowId {}, timestampFrom {}, timestampTo {}, maxCount {}",
-                flowId, timestampFrom, timestampTo, maxCount);
-        if (maxCount < 1) {
+                flowId, constraints.getTimeFrom(), constraints.getTimeTo(), constraints.getMaxCount());
+        if (constraints.getMaxCount() < 1) {
             throw new MessageException(RequestCorrelationId.getId(), System.currentTimeMillis(),
-                    ErrorType.PARAMETERS_INVALID, format("Invalid `max_count` argument '%s'.", maxCount),
+                    ErrorType.PARAMETERS_INVALID, format("Invalid `max_count` argument '%s'.",
+                    constraints.getMaxCount()),
                     "`max_count` argument must be positive.");
         }
```

Comment:
It seams to me that you do not need this check anymore since it exists in the 
FlowHistoryRangeConstraints constructor.

Reply:
Some leftover of refactoring, thanks for spotting! I'll fix it.